src/spikeinterface/postprocessing/unit_localization.py

```diff
@@ -256,27 +265,42 @@ def compute_monopolar_triangulation(waveform_extractor, optimizer='minimize_with
     unit_ids = waveform_extractor.sorting.unit_ids
 
     contact_locations = waveform_extractor.get_channel_locations()
+    nbefore = waveform_extractor.nbefore
 
     sparsity = compute_sparsity(waveform_extractor, method='radius', radius_um=radius_um)
     templates = waveform_extractor.get_all_templates(mode='average')
 
+    if enforce_decrease:
+        neighbours_mask = np.zeros((templates.shape[0], templates.shape[2]), dtype=bool)
+        for i, unit_id in enumerate(unit_ids):
+            chan_inds = sparsity.unit_id_to_channel_indices[unit_id]
+            neighbours_mask[i, chan_inds] = True
+        enforce_decrease_radial_parents = make_radial_order_parents(contact_locations, neighbours_mask)
+        best_channels = get_template_extremum_channel(waveform_extractor, outputs='index')
+
     unit_location = np.zeros((unit_ids.size, 4), dtype='float64')
     for i, unit_id in enumerate(unit_ids):
         chan_inds = sparsity.unit_id_to_channel_indices[unit_id]
         local_contact_locations = contact_locations[chan_inds, :]
 
         # wf is (nsample, nchan) - chann is only nieghboor
-        wf = templates[i, :, :]
-        wf_ptp = wf[:, chan_inds].ptp(axis=0)
-        unit_location[i] = solve_monopolar_triangulation(wf_ptp, local_contact_locations, max_distance_um, optimizer)
+        wf = templates[i, :, :][:, chan_inds]
+        if feature == 'ptp':
+            wf_data = wf.ptp(axis=0)
+        elif feature == 'energy':
+            wf_data = np.linalg.norm(wf, axis=0)
+        #elif feature == 'v_peak':
+        #    wf_data = wf[nbefore]
```

Comment:
@yger since you're at it can you add this too?

Biophysically speaking, this is the best option IMO :)

Reply:
Yes and no, because not sure about this one. At least, I would prevent the use of enforce_decrease, that has the assumption that all values should be decreasing as function of the distance. Or I should consider abs values? Having signed values might disturb everything, but willing to have a try